app/audit/AuditService.scala

```diff
@@ -36,9 +38,19 @@ class AuditService @Inject()(auditConnector: AuditConnector, messageTranslator:
   def authAudit(auditType: AuditType, details: AuthenticationDetails)(implicit hc: HeaderCarrier): Unit =
     auditConnector.sendExplicitAudit(auditType.toString, details)
 
-  def auditEvent(auditType: AuditType, customerId: Ior[TURN, EORINumber], message: Message, channel: ChannelType)(implicit hc: HeaderCarrier): Unit = {
+  def auditEvent(auditType: AuditType, customerId: Ior[TURN, EORINumber], message: Message, channel: ChannelType)(implicit hc: HeaderCarrier): Unit =
+    auditEventWithBox(auditType, customerId, message, channel, None)
+
+  def auditEventWithBox(auditType: AuditType, customerId: Ior[TURN, EORINumber], message: Message, channel: ChannelType, boxOpt: Option[Box])(
+    implicit hc: HeaderCarrier): Unit = {
     val details = AuthenticatedAuditDetails(channel, customerId, messageTranslator.translate(toJson(message.message)))
-    auditConnector.sendExplicitAudit(auditType.toString, details)
+    boxOpt match {
+      case Some(box) =>
+        val detailsWithBox = Json.toJson(details).as[JsObject] + ("boxId" -> JsString(box.boxId.value))
```

Comment:
I'd rather you used a new auditing case class (AuthenticatedAuditDetailsWithBox?) rather than bodging the Json.

Reply:
Yeah, that's fair enough, I'll do that.